symutils/symutils.go

```diff
@@ -264,3 +286,69 @@ func (k *KeyMutex) String() (string, error) {
 
 	return k.Key.String()
 }
+
+type KeySequential struct {
+	k       *KeyAEAD
+	counter uint32
+	salt    []byte
+}
+
+func (k *KeySequential) incrementCounter() {
+	k.counter++
+}
+
+func (k *KeySequential) Encrypt(t []byte, extras ...[]byte) ([]byte, error) {
```

Comment:
can't we get away with just a small diff in the main implementation of KeyAEAD.Encrypt ?
afaict the only difference lies in the counter + nonce management, it looks like a simple switchcase in the code path handles it neatly.